crates/sui-indexer-alt-jsonrpc/src/data/objects.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::HashMap, sync::Arc};
+
+use async_graphql::dataloader::Loader;
+use sui_indexer_alt_schema::objects::StoredObject;
+use sui_types::{base_types::ObjectID, base_types::SequenceNumber};
+
+use super::reader::{ReadError, Reader};
+
+/// Load objects by key (object_id, version).
+#[async_trait::async_trait]
+impl Loader<(ObjectID, SequenceNumber)> for Reader {
+    type Value = StoredObject;
+    type Error = Arc<ReadError>;
+
+    async fn load(
+        &self,
+        keys: &[(ObjectID, SequenceNumber)],
+    ) -> Result<HashMap<(ObjectID, SequenceNumber), Self::Value>, Self::Error> {
+        if keys.is_empty() {
+            return Ok(HashMap::new());
+        }
+
+        let conditions = keys
+            .iter()
+            .map(|key| {
+                format!(
+                    "object_id = '\\x{}' AND object_version = {:?}",
+                    hex::encode(key.0.as_ref()),
+                    key.1
+                )
+            })
+            .collect::<Vec<_>>();
+        let query = format!(
+            "SELECT * FROM kv_objects WHERE {}",
+            conditions.join(" AND ")
```

Comment:
Should this be "OR" instead since we want to fetch multiple objects? Also it might be safer to have parentheses around each condition.

Reply:
Yes good catch!